File: experiments_records/1_latent vulnerabilities/2_requirement_level/python/educollab_flask_req05_repair_pass/models/comment_repo.py
from models.db import get_db


def create_comment(post_id, course_id, author_id, body):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO comments (post_id, course_id, author_id, body) VALUES (?, ?, ?, ?)',
        (post_id, course_id, author_id, body),
    )
    db.commit()
    return get_comment_by_id(course_id, post_id, cursor.lastrowid)



def list_comments(course_id, post_id):
    db = get_db()
    return db.execute(
        'SELECT c.comment_id, c.post_id, c.course_id, c.author_id, c.body, c.created_at, c.updated_at, '
        'u.username, u.display_name '
        'FROM comments c JOIN users u ON u.user_id = c.author_id '
        'WHERE c.course_id = ? AND c.post_id = ? ORDER BY c.comment_id ASC',
        (course_id, post_id),
    ).fetchall()



def get_comment_by_id(course_id, post_id, comment_id):
    db = get_db()
    return db.execute(
        'SELECT c.comment_id, c.post_id, c.course_id, c.author_id, c.body, c.created_at, c.updated_at, '
        'u.username, u.display_name '
        'FROM comments c JOIN users u ON u.user_id = c.author_id '
        'WHERE c.course_id = ? AND c.post_id = ? AND c.comment_id = ?',
        (course_id, post_id, comment_id),
    ).fetchone()



def update_comment(course_id, post_id, comment_id, body):
    db = get_db()
    db.execute(
        'UPDATE comments SET body = ?, updated_at = CURRENT_TIMESTAMP '
        'WHERE course_id = ? AND post_id = ? AND comment_id = ?',
        (body, course_id, post_id, comment_id),
    )
    db.commit()
    return get_comment_by_id(course_id, post_id, comment_id)



def delete_comment(course_id, post_id, comment_id):
    db = get_db()
    db.execute(
        'DELETE FROM comments WHERE course_id = ? AND post_id = ? AND comment_id = ?',
        (course_id, post_id, comment_id),
    )
    db.commit()



def search_comments(course_id, keyword):
    db = get_db()
    term = f'%{keyword}%'
    return db.execute(
        'SELECT c.comment_id, c.post_id, c.course_id, c.author_id, c.body, c.created_at, c.updated_at, '
        'u.username, u.display_name '
        'FROM comments c JOIN users u ON u.user_id = c.author_id '
        'WHERE c.course_id = ? AND c.body LIKE ? '
        'ORDER BY c.comment_id DESC',
        (course_id, term),
    ).fetchall()

File: EduCollab_benchmark/3_file_level_generated_artifacts/javascript/educollab_js_mvc_vul003/src/repos/commentRepo.js
const { db, nowIso } = require('../db');

function createComment({ course_id, post_id, author_id, body }) {
  const ts = nowIso();
  const info = db.prepare(
    'INSERT INTO comments (post_id, course_id, author_id, body, created_at, updated_at) VALUES (?,?,?,?,?,?)'
  ).run(post_id, course_id, author_id, body, ts, ts);
  return getById(course_id, post_id, info.lastInsertRowid);
}

function listComments(course_id, post_id) {
  return db.prepare(`
    SELECT c.*, u.username as author_username
    FROM comments c
    JOIN users u ON u.user_id = c.author_id
    WHERE c.course_id=? AND c.post_id=?
    ORDER BY c.comment_id ASC
  `).all(course_id, post_id);
}

function getById(course_id, post_id, comment_id) {
  return db.prepare(`
    SELECT c.*, u.username as author_username
    FROM comments c
    JOIN users u ON u.user_id = c.author_id
    WHERE c.course_id=? AND c.post_id=? AND c.comment_id=?
  `).get(course_id, post_id, comment_id);
}

function updateComment(course_id, post_id, comment_id, body) {
  db.prepare('UPDATE comments SET body=?, updated_at=? WHERE course_id=? AND post_id=? AND comment_id=?')
    .run(body, nowIso(), course_id, post_id, comment_id);
  return getById(course_id, post_id, comment_id);
}

function deleteComment(course_id, post_id, comment_id) {
  db.prepare('DELETE FROM comments WHERE course_id=? AND post_id=? AND comment_id=?')
    .run(course_id, post_id, comment_id);
}

function searchComments(course_id, keyword) {
  const k = `%${keyword}%`;
  return db.prepare(`
    SELECT c.*, u.username as author_username
    FROM comments c
    JOIN users u ON u.user_id = c.author_id
    WHERE c.course_id=? AND c.body LIKE ?
    ORDER BY c.comment_id DESC
  `).all(course_id, k);
}

module.exports = { createComment, listComments, getById, updateComment, deleteComment, searchComments };
